### PR TITLE
Support watchers for github issue labels

### LINKED ISSUE #84: Support watchers for github issue labels
For people who don't want to watch all the Servo issue activity but still care about activity that goes on in particular labels, highfive could support automatically notifying them when a label is added to an issue. This would be implemented very similarly to the [file watching handler](https://github.com/servo/highfive/tree/master/handlers/watchers) but using the "labeled" action, which would need to be added to `eventhandler.py`.

### PR DESCRIPTION
**Pull request by canaltinova**
Fixes #84 .
Created a test for watched and a test for not watched labels.

### CODE REVIEW

File: handlers/label_watchers/__init__.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        try:
+            watchers = config.items(repo)
+        except ConfigParser.NoSectionError:
+            return # No watchers
+
+        mentions = []
+        for (watcher, watched_labels) in watchers:
+            watched_labels = watched_labels.split(' ')
+
+            for watched_label in watched_labels:
```
Comment (jdm) [inline handlers/label_watchers/__init__.py]: This loop can be replaced by `if new_label in watched_labels`.

File: handlers/label_watchers/__init__.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        config = get_label_config()
+
+        repo = api.owner + '/' + api.repo
+        try:
+            watchers = config.items(repo)
+        except ConfigParser.NoSectionError:
+            return # No watchers
+
+        mentions = []
+        for (watcher, watched_labels) in watchers:
```
Comment (jdm) [inline handlers/label_watchers/__init__.py]:
```py
if watcher == payload['sender']['login']:
    continue
```
and add a test for this, please.

**Comment by jdm** [comment]
This looks great! One small change to avoid redundant notifications and it should be good to merge!

**Comment by canaltinova** [comment]
Thanks for the review. Created a new commit!

**Comment by jdm** [comment]
Thanks!

**Pull request merged**